scipy/stats/_stats_py.py

```diff
@@ -9440,3 +9442,98 @@ def rankdata(a, method='average', *, axis=None, nan_policy='propagate'):
         result[nan_indexes] = np.nan
 
     return result
+
+
+def expectile(a, alpha=0.5, *, dtype=None, weights=None):
+    r"""Compute the expectile along the specified axis.
+
+
+    Expectiles are a generalization of the expectation in the same way as
+    quantiles are a generalization of the median. The expectile at level
+    `alpha = 0.5` is the mean (average). See Notes for more details.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    alpha : float
+        The level of the expectile; `alpha=0.5` gives the mean.
+    dtype : dtype, optional
+        Type to which the input arrays are cast before the calculation is
+        performed.
+    weights : array_like, optional
+        The `weights` array must be broadcastable to the same shape as `a`.
+        Default is None, which gives each value a weight of 1.0.
+
+    Returns
+    -------
+    expectile : ndarray
+        See `dtype` parameter above.
+
+    See Also
+    --------
+    numpy.mean : Arithmetic average
+    numpy.quantile : Quantile
+
+    Notes
+    -----
+    The expectile at level `alpha` (:math:`\alpha`) of the array `a`
+    (:math:`a_i`) associated to `weights` (:math:`w_i`) is the unique solution
+    :math:`t` of:
+
+    .. math::
+
+        \alpha \sum_{i=1}^n w_i (a_i - t)_+ =
+            (1 - \alpha) \sum_{i=1}^n w_i (t - a_i)_+ \,.
+
+    References
+    ----------
+    .. [1] W. K. Newey and J. L. Powell (1987), "Asymmetric Least Squares
+           Estimation and Testing," Econometrica, 55, 819-847.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.stats import expectile
+    >>> a =[1, 4, 2, -1]
+    >>> expectile(a, alpha=0.5) == np.mean(a)
+    True
+    >>> expectile(a, alpha=0.2)
+    0.42857142857142855
+    >>> expectile(a, alpha=0.8)
+    2.5714285714285716
+
+    """
+    if alpha < 0 or alpha > 1:
```

Comment:
would it be worth special-casing `alpha=0` & `=1` (genuine question, answer might be no)

Reply:
2 view points here:
- implementation: yes, because we then know the answer in terms of min/max.
- testing: no, because it's then hard(er) to honestly test the implementation at the boundaries.

Reply:
I understand the trade-off. I don't have a strong opinion but it does seem "wasteful" to me not to short-circuit those cases here...

Reply:
Also no strong opinion on my side. The boundary alpha=0 and 1 is just not that interesting (same for quantiles).